test/TesterInternal/OrleansRuntime/Streams/FixedSizeObjectPoolTests.cs

```diff
@@ -11,50 +10,38 @@ public class FixedSizeObjectPoolTests
     {
         private class Accumulator
         {
-            public int CurrentlyAllocated { get; set; }
             public int MaxAllocated { get; set; }
 
         }
         private class TestPooledResource : PooledResource<TestPooledResource>
         {
-            private readonly Accumulator accumulator;
-
-            public TestPooledResource(IObjectPool<TestPooledResource> pool, Accumulator accumulator)
-                : base(pool)
+            public TestPooledResource(Accumulator accumulator)
             {
-                this.accumulator = accumulator;
-                this.accumulator.CurrentlyAllocated++;
-                this.accumulator.MaxAllocated = Math.Max(this.accumulator.MaxAllocated, this.accumulator.CurrentlyAllocated);
-            }
-
-            public override void OnResetState()
-            {
-                this.accumulator.CurrentlyAllocated--;
+                accumulator.MaxAllocated++;
             }
         }
 
         [Fact, TestCategory("BVT"), TestCategory("Streaming")]
         public void Alloc1Free1Test()
         {
             var accumulator = new Accumulator();
-            IObjectPool<TestPooledResource> pool = new FixedSizeObjectPool<TestPooledResource>(10, p => new TestPooledResource(p, accumulator));
+            IObjectPool<TestPooledResource> pool = new FixedSizeObjectPool<TestPooledResource>(10, () => new TestPooledResource(accumulator));
 
             // Allocate and free 10 items
-            for (int i = 0; i < 10; i++)
+            for (int i = 0; i < 20; i++)
```

Comment:
may be make this dependent on the fixed pool size used? (also the comment and the value no longer match)